galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/notif/VErrorNotification.kt

```diff
@@ -18,84 +18,59 @@
 import com.vaadin.flow.component.button.Button
-import com.vaadin.flow.component.details.Details
-import com.vaadin.flow.component.details.DetailsVariant
-import com.vaadin.flow.component.html.H3
-import com.vaadin.flow.component.html.Span
-import com.vaadin.flow.component.orderedlayout.FlexComponent
+import com.vaadin.flow.component.icon.VaadinIcon
 
 /**
  * Error type notification component.
+ *
+ * @param title the error notification title.
+ * @param message the error notification message.
  */
-open class VErrorNotification(title: String?, message: String) : VAbstractNotification() {
+class VErrorNotification(title: String?,
+                         message: String,
+                         locale: String)
+  : VAbstractNotification(title, message, locale) {
 
   //-------------------------------------------------
   // IMPLEMENTATION
   //-------------------------------------------------
 
-  /**
-   * Creates the error notification footer.
-   */
-  override fun createFooter() {
-    footer.add(details)
-    footer.add(close)
-    footer.isSpacing = true
-    footer.justifyContentMode = FlexComponent.JustifyContentMode.CENTER
-    footer.setVerticalComponentAlignment(FlexComponent.Alignment.BASELINE, details)
-    footer.setVerticalComponentAlignment(FlexComponent.Alignment.BASELINE, close)
-    super.setFooter(footer)
-  }
-
-  override fun setButtons(locale: String) {
-    close = VInputButton(LocalizedProperties.getString(locale, "CLOSE"))
-    if (locale == "fr_FR") {
-      details = Details("Afficher les détails",
-                        Text("Les détails ici"))
-    } else if (locale == "en_GB") {
-      details = Details("Show details",
-                        Text("Details here"))
-    }
-    close.addClickListener { hide() }
-    close.width = "20%"
-    close.height = "50%"
+  override fun setButtons() {
+    close = Button(LocalizedProperties.getString(locale, "CLOSE"))
+    close.addClickListener { close() }
+    close.isAutofocus = true
+    buttons.add(close)
   }
 
-  override fun onEnterEvent(keyDownEvent: ShortcutEvent?) {
-    close.click()
-  }
+  override val iconName: VaadinIcon
+    get() = VaadinIcon.EXCLAMATION_CIRCLE
 
-  override fun onRightEvent(keyDownEvent: ShortcutEvent?) {
-    TODO("Not yet implemented")
+  fun onArrowUpEvent(keyDownEvent: ShortcutEvent?) {
+    close.focus()
   }
 
-  override fun onLeftEvent(keyDownEvent: ShortcutEvent?) {
-    TODO("Not yet implemented")
+  fun onArrowDownEvent(keyDownEvent: ShortcutEvent?) {
+    details.focus()
   }
 
   //--------------------------------------------------
   // DATA MEMBERS
   //--------------------------------------------------
-  var details = Details()
-  var close = VInputButton()
-  val listener: ComponentEventListener<ClickEvent<Button>>? = null
-  override val iconName: String
-    get() = "warning"
+  private var details = ErrorMessageDetails(message, locale, this)
+  private lateinit var close: Button
```

Comment:
can we put here `private val close = = Button(LocalizedProperties.getString(locale, "CLOSE"))`